apps/web/middleware.ts

```diff
@@ -41,21 +41,9 @@ const middleware = async (req: NextRequest): Promise<NextResponse<unknown>> => {
     }
   }
 
-  const res = routingForms.handle(url);
-
-  const { nonce } = csp(req, res ?? null);
-
-  if (!process.env.CSP_POLICY) {
-    req.headers.set("x-csp", "not-opted-in");
-  } else if (!req.headers.get("x-csp")) {
-    // If x-csp not set by gSSP, then it's initialPropsOnly
-    req.headers.set("x-csp", "initialPropsOnly");
-  } else {
-    req.headers.set("x-csp", nonce ?? "");
-  }
```

Comment:
This is how we handle CSP with App Router but these headers were supposed to be on res. So, I have fixed it now.